internal/monitor/reader.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/monitor/tags"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"golang.org/x/net/context"
+)
+
+var (
+	// When a first read call is made by the user, we either fetch entire file or x number of bytes from GCS based on the request.
+	// Now depending on the pagesize multiple read calls will be issued by user to read the entire file. These
+	// requests will be served from the downloaded data.
+	// This metric captures only the requests made to GCS, not the subsequent page calls.
+	gcsReadCount = stats.Int64("gcs/read_count",
+		"Specifies the count of gcs reads made along with type",
+		stats.UnitDimensionless)
+	downloadBytesCount = stats.Int64("gcs/download_bytes_count",
+		"Cumulative number of bytes downloaded from GCS along with read type",
+		stats.UnitBytes)
+	fileCacheReadCount = stats.Int64("file_cache/read_count",
+		"Specifies the number of read requests made via file cache along with type - Sequential/Random and cache hit - true/false",
+		stats.UnitDimensionless)
+	fileCacheReadBytesCount = stats.Int64("file_cache/read_bytes_count",
+		"The cumulative number of bytes read from file cache along with read type - Sequential/Random",
+		stats.UnitBytes)
+	fileCacheReadLatency = stats.Float64("file_cache/read_latency",
+		"Latency of read from file cache along with cache hit - true/false",
+		stats.UnitMilliseconds)
+)
+
+const NanosecondsInOneMillisecond = 1000000
+
+// Initialize the metrics.
+func init() {
+	// GCS related metrics
+	if err := view.Register(
+		&view.View{
+			Name:        "gcs/read_count",
+			Measure:     gcsReadCount,
+			Description: "Specifies the number of gcs reads made along with type- Sequential/Random",
+			Aggregation: view.Sum(),
+			TagKeys:     []tag.Key{tags.ReadType},
+		},
+		&view.View{
+			Name:        "gcs/download_bytes_count",
+			Measure:     downloadBytesCount,
+			Description: "The cumulative number of bytes downloaded from GCS.",
```

Comment:
nit: The cumulative number of bytes downloaded from GCS along with type - Sequential/Random